### PR TITLE
List performance 

### PR DESCRIPTION
**Pull request by ffugly**
Implemented fetchListIfNeeded as described in the (redux docs)[http://redux.js.org/docs/advanced/AsyncActions.html#async-action-creators] plus a few extra tweaks learned from managing large data sets.

### CODE REVIEW

File: client/modules/post/actions/postListActions.js [hunk 1]
```diff
@@ -89,10 +116,94 @@ function receiveFeaturedPostList(json) {
...
+  }
+  if(state.post.list.isFetching) {
+    console.log("FETCHING ALREADY DON'T FETCH LIST");
+    return false
+  }
+  return state.post.list.didInvalidate[postType]
+}
+
+
+export const fetchListIfNeeded = (postType) => (dispatch, getState) => {
```
Comment (grantfowler) [inline client/modules/post/actions/postListActions.js:158]: how do we pass variables in to this? a lot of our soflete stuff will be "fetchListBySquad(squadId)" for example

Comment (grantfowler) [reply client/modules/post/actions/postListActions.js:158]: really easy to make it two arguments with the second one optional actually

File: client/modules/post/reducers/postListReducers.js [hunk 2]
```diff
@@ -5,14 +5,30 @@ LIST REDUCERS GO HERE
...
+    all: false
+    , featured: false
+    , published: false
+  }
   , all: []
+  , allItemsMap: {}
   , featured: []
+  , featuredMap: {}
   , published: []
+  , publishedMap: {}
```
Comment (grantfowler) [inline client/modules/post/reducers/postListReducers.js:22]: the actions are all gravy but I think we need to reorganize the store state a little bit. each list type should have its own fetched, updated, and pagination.

File: client/modules/post/actions/postListActions.js (hunk 1 above)
Comment (ffugly) [reply client/modules/post/actions/postListActions.js:158]: yeah, two vars is the way to go there.  

```
export const fetchListIfNeeded = (listType, typeId) => (dispatch, getState) => { 
... 
```

File: client/modules/post/reducers/postListReducers.js (hunk 2 above)
Comment (ffugly) [reply client/modules/post/reducers/postListReducers.js:22]: So should it be something like this? 

``` json
...
   , all: []
  , allItemsMap: {}
  , allPagination: {}
  , allLastUpdated: null 
  , allFilter: ''
  , allSortBy: 'name'
  , allQuery: '' 
  , featured: []
  , featuredMap: {}
  , featuredPagination: {}
  , featuredLastUpdated: null 
  , featuredFilter: ''
  , featuredSortBy: 'name'
  , featuredQuery: '' 
  , published: []
  , publishedMap: {}
  , publishedPagination: {}
  , publishedLastUpdated: null 
  , publishedFilter: ''
  , publishedSortBy: 'name'
  , publishedQuery: '' 
  ...
```

**Pull request merged**